include/llama.h

```diff
@@ -902,13 +909,98 @@ extern "C" {
     // Otherwise, llama_batch.token will be allocated to store n_tokens llama_token
     // The rest of the llama_batch members are allocated with size n_tokens
     // All members are left uninitialized
-    LLAMA_API struct llama_batch llama_batch_init(
+    DEPRECATED(LLAMA_API struct llama_batch llama_batch_init(
+                    int32_t n_tokens,
+                    int32_t embd,
+                    int32_t n_seq_max), "use llama_batch_ext_init instead");
+
+    // Frees a batch of tokens allocated with llama_batch_init()
+    DEPRECATED(LLAMA_API void llama_batch_free(struct llama_batch batch),
+            "use llama_batch_ext API instead");
+
+    // Allocates a batch of tokens on the heap that can hold a maximum of n_tokens
+    // Each token can be assigned up to n_seq_max sequence ids
+    // The batch has to be freed with llama_batch_ext_free()
+    LLAMA_API struct llama_batch_ext * llama_batch_ext_init(
             int32_t n_tokens,
-            int32_t embd,
             int32_t n_seq_max);
 
-    // Frees a batch of tokens allocated with llama_batch_init()
-    LLAMA_API void llama_batch_free(struct llama_batch batch);
+    // Same with llama_batch_init, but initializes the batch with the provided text tokens
+    // First token will be at position pos0
+    // The sequence ID will be fixed to seq_id
+    // If output_last is true, the last token will have output set
+    // The batch has to be freed with llama_batch_ext_free()
+    LLAMA_API struct llama_batch_ext * llama_batch_ext_init_from_text(
+            llama_token * tokens,
+                int32_t   n_tokens,
+                int32_t   pos0,
+                int32_t   seq_id,
+                   bool   output_last);
```

Comment:
Since this call is basically a combination of other calls in the API, we should probably remove it in order to keep the API smaller. Moreover, we can use the C++ version `llama_batch_ext_ptr::init_from_text()`. The C++ API in `llama-cpp.h` can absorb most of the convenience API calls since it supports argument overloading.

Reply:
Hmm yeah that makes sense. Indeed, I aimed to have this C API to be exposed to downstream binding, but on second thought it's better to keep the API minimal

Reply:
Removed in https://github.com/ggml-org/llama.cpp/pull/11875/commits/30f1db9936112bc04917df1a9520421ae5f3ffc1